server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -160,7 +161,12 @@ public InputStream downloadBlob(Iterable<String> path, String fileName) throws I
 
     @Override
     public void deleteBlobs(Iterable<String> path, List<String> fileNames) throws IOException {
-        blobStore.blobContainer((BlobPath) path).deleteBlobsIgnoringIfNotExists(fileNames);
+        if (blobStore.blobContainer((BlobPath) path) instanceof SoftDeleteSupportedBlobContainer) {
+            ((SoftDeleteSupportedBlobContainer) blobStore.blobContainer((BlobPath) path)).softDeleteBlobsIgnoringIfNotExists(fileNames);
+            logger.info("Yay soft deelted");
```

Comment:
We will be changing this, right? Also, typo in `deleted` :)

Reply:
yes, this is just debug logs. removed it now .